staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go

```diff
@@ -275,6 +277,18 @@ func (az *Cloud) DeletePublicIP(service *v1.Service, pipResourceGroup string, pi
 	if rerr != nil {
 		klog.Errorf("PublicIPAddressesClient.Delete(%s) failed: %s", pipName, rerr.Error().Error())
 		az.Event(service, v1.EventTypeWarning, "DeletePublicIPAddress", rerr.Error().Error())
+
+		errMsgCode, err := extractRetryErrorMsgCode(rerr.Error().Error())
+		if err != nil {
+			klog.Errorf("DeletePublicIP for public IP %s failed when extracting error message code with error %s", pipName, err.Error())
+			return err
+		}
+
+		if strings.EqualFold(cannotDeletePublicIPErrorMessageCode, errMsgCode) {
```

Comment:
use error message matching may be more robust, e.g. match `PublicIPAddressCannotBeDeleted` word, otherwise if got another not formatted error, it would return error just because `extractRetryErrorMsgCode` return error.

Reply:
agreed. could you use 'strings.Contains' here?